### PR TITLE
New Entry: Java Queue

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

New entry for Java Queue. From the earlier collections PR.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/queue/queue.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - 'Queues'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+A `Queue` is an interface provided in `java.util` that extends collections to provide additional methods to access or manipulate items at the head of the queue. Where the "head" of the queue is will be defined by the particular implementation. In a typical implementation, the queue will access elements in a FIFO (first-in-first-out) manner. However, in an stack implementation, the order will be LIFO (last-in-first-out). In a priority queue implementation, the first item out will be defined by its specified priority.
+
+The `Queue` interface utilizes the following methods:
```
Comment (KTom101) [inline content/java/concepts/queue/queue.md:17]: Line 17 and below can have their own subheading.

File: content/java/concepts/queue/queue.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- `.add()` : `.add(item)` adds `item` to the `Queue` if possible, otherwise it throws an exception.
+- `.offer()` : `.offer(item)` adds `item` to the `Queue` if possible, otherwise it returns `false`.
+- `.remove()` : removes and returns the head item of the `Queue`, throwing an exception when the `Queue` is empty.
+- `.poll()` : removes and returns the head item of the `Queue`, returning `null` if the `Queue` is empty.
+- `.element()` : returns the head of the `Queue` without removing it.
+- `.peek()` : returns the head of the `Queue` without removing it.
```
Comment (KTom101) [inline content/java/concepts/queue/queue.md:24]: Can a couple of these methods be shown being used on a Queue?

**Review by KTom101 — CHANGES_REQUESTED** [review]
@SSwiniarski I've added some comments to this.

**Comment by SSwiniarski** [comment]
@KTom101, I implemented your suggestions by adding an example using a LinkedList.

**Review by KTom101 — APPROVED** [review]

File: content/java/concepts/queue/queue.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  - 'Queues'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+A `Queue` is an interface provided in the `java.util` package that extends collections to provide additional methods to access or manipulate items at the head of the queue. Where the "head" of the queue is defined by the particular implementation. In a typical implementation, the queue will access elements in a FIFO (first-in-first-out) manner. However, in a stack implementation, the order will be LIFO (last-in-first-out). In a priority queue implementation, the first item out will be defined by its specified priority.
+
+## Mehods
```
Comment (Dusch4593) [inline content/java/concepts/queue/queue.md:17]: I think we should move this section somewhere below the "Syntax" section.

File: content/java/concepts/queue/queue.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+The `Queue` interface utilizes the following methods:
+
+- `.add()` : `.add(item)` adds `item` to the `Queue` if possible, otherwise it throws an exception.
+- `.offer()` : `.offer(item)` adds `item` to the `Queue` if possible, otherwise it returns `false`.
+- `.remove()` : removes and returns the head item of the `Queue`, throwing an exception when the `Queue` is empty.
+- `.poll()` : removes and returns the head item of the `Queue`, returning `null` if the `Queue` is empty.
+- `.element()` : returns the head of the `Queue` without removing it.
+- `.peek()` : returns the head of the `Queue` without removing it.
+
+## Syntax
```
Comment (Dusch4593) [inline content/java/concepts/queue/queue.md:28]: But this seems more like an example section? Could include both a "Syntax" and "Example" section for this?

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Finished with my review for this one and left a few comments/suggestions.

File: content/java/concepts/queue/queue.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
+---
+Title: 'Queue'
+Description: 'A collection interface offering additional operations to access or manipulate items at the head of the queue.'
```
[the remark this answers is not in the document]
Comment (SSwiniarski) [reply content/java/concepts/queue/queue.md:3]: A Queue interface is not inherently FIFO.

**Comment by SSwiniarski** [comment]
@Dusch4593, I implemented most of your suggestions.

**Review by Dusch4593 — APPROVED** [review]
@SSwiniarski LGTM! 👍🏻 Made a super small edit in 0a7246a0f9e80ecd3c35091d04ed949ee41febc0. This seems good for merge. 😄

**Pull request merged**